sound/soc/intel/boards/sof_sdw.c

```diff
@@ -995,12 +995,29 @@ static int mc_probe(struct platform_device *pdev)
 	return ret;
 }
 
+static int mc_remove(struct platform_device *pdev)
+{
+	int ret;
+	int i;
+
+	for (i = 0; i < ARRAY_SIZE(codec_info_list); i++)
+		if (codec_info_list[i].exit && codec_info_list[i].is_probed) {
```

Comment:
could we instead do the is_probed() checks inside of the .exit() callback?
in other words call .exit() unconditionally and check inside each callback implementation if the resources being released were actually allocated in the first place. This might even allow us to remove this flag?

Reply:
@plbossart I think we still need a flag to record the status when we check it inside the .exit function. Now, I use dai_link as an argument of .exit function so we can get codec name from dai_link and we won't call the .exit function if we don't find the dai_link.